pkg/objstore/swift/swift.go

```diff
@@ -82,215 +190,130 @@ func (c *Container) Name() string {
-	res := objects.Create(c.client, c.name, name, options)
-	return res.Err
-}
-
-// Delete removes the object with the given name.
-func (c *Container) Delete(ctx context.Context, name string) error {
-	return objects.Delete(c.client, c.name, name, nil).Err
-}
-
-func (*Container) Close() error {
-	// Nothing to close.
-	return nil
-}
-
-func parseConfig(conf []byte) (*SwiftConfig, error) {
-	var sc SwiftConfig
-	err := yaml.UnmarshalStrict(conf, &sc)
-	return &sc, err
-}
-
-func authOptsFromConfig(sc *SwiftConfig) gophercloud.AuthOptions {
-	authOpts := gophercloud.AuthOptions{
-		IdentityEndpoint: sc.AuthUrl,
-		Username:         sc.Username,
-		UserID:           sc.UserId,
-		Password:         sc.Password,
-		DomainID:         sc.DomainId,
-		DomainName:       sc.DomainName,
-		TenantID:         sc.ProjectID,
-		TenantName:       sc.ProjectName,
-
-		// Allow Gophercloud to re-authenticate automatically.
-		AllowReauth: true,
+func (c *Container) Upload(_ context.Context, name string, r io.Reader) error {
+	size, err := objstore.TryToGetSize(r)
+	if err != nil {
+		level.Warn(c.logger).Log("msg", "could not guess file size, using large object to avoid issues if the file is larger than limit", "name", name, "err", err)
+		// Anything higher or equal to chunk size so the SLO is used.
+		size = c.chunkSize
 	}
-
-	// Support for cross-domain scoping (user in different domain than project).
-	// If a userDomainName or userDomainID is given, the user is scoped to this domain.
-	switch {
-	case sc.UserDomainName != "":
-		authOpts.DomainName = sc.UserDomainName
-	case sc.UserDomainID != "":
-		authOpts.DomainID = sc.UserDomainID
+	var file io.WriteCloser
+	if size >= c.chunkSize {
+		file, err = c.connection.StaticLargeObjectCreateFile(&swift.LargeObjectOpts{
+			Container:        c.name,
+			ObjectName:       name,
+			ChunkSize:        c.chunkSize,
+			SegmentContainer: c.segmentsContainer,
+			CheckHash:        true,
+		})
+	} else {
+		file, err = c.connection.ObjectCreate(c.name, name, true, "", "", swift.Headers{})
 	}
-
-	// A token can be scoped to a domain or project.
-	// The project can be in another domain than the user, which is indicated by setting either projectDomainName or projectDomainID.
-	switch {
-	case sc.ProjectDomainName != "":
-		authOpts.Scope = &gophercloud.AuthScope{
-			DomainName: sc.ProjectDomainName,
-		}
-	case sc.ProjectDomainID != "":
-		authOpts.Scope = &gophercloud.AuthScope{
-			DomainID: sc.ProjectDomainID,
-		}
+	if err != nil {
+		return errors.Wrap(err, "swift failed to create file")
 	}
-	if authOpts.Scope != nil {
-		switch {
-		case sc.ProjectName != "":
-			authOpts.Scope.ProjectName = sc.ProjectName
-		case sc.ProjectID != "":
-			authOpts.Scope.ProjectID = sc.ProjectID
-		}
+	defer runutil.CloseWithLogOnErr(c.logger, file, "swift upload obj close")
+	if _, err := io.Copy(file, r); err != nil {
+		return errors.Wrap(err, "failed to write uploaded file to swift")
```

Comment:
```suggestion
		return errors.Wrap(err, "downloading swift object")
```

Reply:
hm, there is uploading going on? so maybe `uploading swift object` ?  but otherwise :+1: from me for the suggestion

Reply:
changed here https://github.com/thanos-io/thanos/pull/2732/commits/7a51f585a7184cc9a0c5a73ffcd77b23896da026 to `uploading swift object`, hope that's ok

Reply:
still not changed here though

Reply:
sorry, should be fixed now :crossed_fingers: